REVIEW_COPY.md

```diff
@@ -0,0 +1,312 @@
+
+```elm
+-- In this example a time slot always covers whole hours,
+-- but we could make this minutes or seconds if we wanted.
+type alias Model =
+    { from : Hour, until : Hour }
+
+type alias Hour =
+    Int
+```
+
+Now for the drag & drop behavior. In our example all you can do is drag left-to-right. This means that when the user presses down we store the hour the cursor is over as the left bound of the time slot. Then as the cursor moves to the right we will update the `until` time of the timeslot, until the user releases.
+
+We will need to know which our the cursor is over. We can calculate this if we know the position of the cursor and the position and dimensions of the slider on the screen. Lets be optimistic and assume we just get that information. Our `Msg` type we design like this:
+
+```elm
+type alias Msg =
+    { event : DragEvent
+    , cursor : Coords
+    , sliderPosition : Rect
+    }
+
+type DragEvent
+    = Start
+      -- We don't need to do anything special on a Stop event, so we can treat
+      -- it the same as a Move event.
+    | MoveOrStop
+
+type alias Coords =
+    { x : Float, y : Float }
+
+type alias Rect =
+    { x : Float, y : Float, width : Float, height : Float }
+```
+
+Now we can proof we are able to calculate the hour the cursor is over if we get `Msg`s that look like the above.
+
+```elm
+cursorAtHour : Msg -> Hour
+cursorAtHour { cursor, sliderPosition } =
+    let
+        dx =
+            cursor.x - sliderPosition.x
+
+        atMost =
+            min
+
+        atLeast =
+            max
+    in
+    (24 * (dx / sliderPosition.width))
+        |> floor
+        -- Ensure we get a number between 0 and 23,
+        -- even if the cursor moves to the left or right of the slider.
+        |> atMost 23
+        |> atLeast 0
+```
+
+All that's left to do is use this in our `update` function. When we get a `Start` event we update the `from` field in the model, and when we get a `MoveOrStop` event the `until` field.
+
+```elm
+update : Msg -> Model -> Model
+update msg model =
+    let
+        hour =
+            cursorAtHour msg
+    in
+    case msg.event of
+        Start ->
+            if coordsInRect msg.cursor msg.sliderPosition then
+                { from = hour
+                , until = hour + 1
+                }
+
+            else
+                model
+
+        MoveOrStop ->
+            { model
+                | until = 1 + max hour model.from
+            }
+
+coordsInRect : Coords -> Rect -> Bool
+coordsInRect =
+    Debug.todo "Implementation omitted for brevity."
+```
+
+And we have an app! Well almost, we've not discussed where these `Msg`s will be coming from. There's a bit of JavaScript responsible for that, which I'll talk more about in a bit. If you want to peak ahead check out the [time slider source code].
```

Comment:
little typo
```suggestion
And we have an app! Well almost, we've not discussed where these `Msg`s will be coming from. There's a bit of JavaScript responsible for that, which I'll talk more about in a bit. If you want to peek ahead check out the [time slider source code].
```

(p.s. there was a great Twitter account about this particular common typo: https://twitter.com/stealthmountain)